internal/pkg/cli/deploy/workload.go

```diff
@@ -339,19 +339,27 @@ func (img ContainerImageIdentifier) customOrGitTag() string {
 	return img.GitShortCommitTag
 }
 
+// sidecarReferenceTag returns the tag that should be used to reference the image.
+func (img ContainerImageIdentifier) sidecarReferenceTag() string {
```

Comment:
As per the design proposal we decided not to tag Sidecar images with `custom tag` provided by user, so that is the reason for separate method to tag sidecar images instead of using `ReferenceTag()`.